npalm/terraform-aws-gitlab-runner
# feat: add runners_pull_policies to support multiple pull policies

## Description

Support multiple Docker pull policies. The pull_policy parameter allows to specify a list of pull policies. The policies in the list will be attempted in order from left to right until a pull attempt is successful, or the list is exhausted.

https://docs.gitlab.com/runner/executors/docker.html#how-pull-policies-work

## Migrations required

NO - A deprecation notice was added to description. In a future release the variable runners_pull_policy could be removed in favor of runners_pull_policies.

## Verification

I have verified the examples (https://github.com/gmeligio/terraform-aws-gitlab-runner/actions/runs/3040828783):
- Verify examples (1.0.11, runner-default)
- Verify examples (1.0.11, runner-docker)
- Verify examples (1.0.11, runner-multi-region)
- Verify examples (1.0.11, runner-pre-registered)
- Verify examples (1.0.11, runner-public)
- Verify examples (latest, runner-default)
- Verify examples (latest, runner-docker)
- Verify examples (latest, runner-multi-region)
- Verify examples (latest, runner-pre-registered)
- Verify examples (latest, runner-public)
